### DISCUSSION TITLE
Making this lib comparable to HotChocolate in terms of features

### DISCUSSION THREAD

**Discussion by kasir-barati** [TriPSs/nestjs-query#333] [Ideas]
Hey folks,

I wanted to see if there is any plan for how to develop and add new features to this lib. To be perfectly honest with you I was at first disappointed that we do not have something as capable as [HotChocolate](https://chillicream.com/docs/hotchocolate/v13). But then I found this lib and I am really happy about it since now we can find our way towards a feature-rich GraphQL server with ease by leveraging this library.

I know it feels like a long shot but I think it is possible. On another note I am planning to use this lib extensively in my personal or work related projects. So I am already on-board, the only thing that I lack at the moment are:
- A bit of insight into this lib intricacies (Any comment or pointer would be much appreciated :slightly_smiling_face:). But I am also working on it.
- I migrated from RESTful API landscape to GraphQL. So I am relatively new to this setup and still needs to learn a lot. But I have been using GraphQL already a lot lately and going forward my exposure to it is just going to increase.

Any comment?

**Comment by TriPSs** [comment]
Always open on PRs/feedback/ideas to improve this lib, so everything is welcome!

**Reply by kasir-barati** [reply]
## Some housekeeping todos that IMO are [broken windows](https://rafalkozlowski.engineer/broken-glass-theory-understanding-the-consequences-of-technical-debt-in-software-engineering/)

- Very long lines of code which makes it kinda hard to understand.
- `moduleResolution` should be changed to `NodeNext`.
- Change the hard to understand names like `DM`, `DOSA` which frankly I can kinda figure them out but nonetheless it is not very intuitive. You need to pause and poke around to understand them.

## Kinda Falls Under the Category of Nitpicky

- `jest.e2e.js` can go inside the `examples` dir.
- ~Update the `compose` file syntax (remove `version` from it, rename it to `compose.yaml`).~
- Maybe `pnpm` instead of yarn for the sake of faster installation and needing less space on our local machines.

## PRs

1. https://github.com/TriPSs/nestjs-query/pull/337
2. https://github.com/TriPSs/nestjs-query/pull/343
3. https://github.com/TriPSs/nestjs-query/pull/346
4. https://github.com/TriPSs/nestjs-query/pull/347

**Comment by kasir-barati** [comment]
Sorry @TriPSs to bug you here. But I would love to see if we can keep prettier: https://www.reddit.com/r/node/comments/w95n80/comment/ihtec8z/?utm_source=share&utm_medium=web3x&utm_name=web3xcss&utm_term=1&utm_content=share_button

And if you're onboard I this we can talk about the line length to something like 70 since right now it is a bit hard for me to read the code.

**Reply by kasir-barati** [reply]
@TriPSs If you can make time to review a big PR for it I can open a PR for it and apply prettier to the whole repo. Let's discuss the line width later.

**Reply by kasir-barati** [reply]
@TriPSs I guess I have found what you mentioned for eslint: https://prettier.io/docs/en/integrating-with-linters

Though it still needs prettier.
